pymc3/tests/test_theanof.py

```diff
@@ -74,3 +77,23 @@ def gen():
         np.testing.assert_equal(np.ones((10, 10)) * 0, f())
         np.testing.assert_equal(np.ones((10, 10)) * 1, f())
         self.assertRaises(StopIteration, f)
+
+    def test_pickling(self):
+        data = np.random.uniform(size=(1000, 10))
+        minibatches = DataSampler(data, batchsize=50)
+        gen = generator(minibatches)
+        pickle.loads(pickle.dumps(gen))
+        bad_gen = generator(integers())
+        self.assertRaises(Exception, pickle.dumps, bad_gen)
+
+    def test_gen_cloning_with_shape_change(self):
+        data = np.random.uniform(size=(1000, 10))
```

Comment:
Wondering (and this is not necessarily a blocker) if some property tests would be interesting here. I think @AustinRochford built some tests like that. 

Just an off the hand the comment. The rest of the tests look fine to me. I've learned a bit from reading your code!